app/src/main/java/com/example/capstoneproject/ui/main/MainActivity.kt

```diff
@@ -33,27 +35,45 @@ class MainActivity : AppCompatActivity() {
 
         navController.addOnDestinationChangedListener { _, destination, _ ->
             when (destination.id) {
-                R.id.signUpFragment, R.id.signInFragment -> binding.bottomNavigation.visibility = View.GONE
+                R.id.signUpFragment, R.id.signInFragment, R.id.firstFragment -> binding.bottomNavigation.visibility =
+                    View.GONE
+
                 else -> binding.bottomNavigation.visibility = View.VISIBLE
             }
         }
 
         binding.bottomNavigation.setOnItemSelectedListener { item ->
             when (item.itemId) {
                 R.id.signInFragment -> {
-                    signOut()
+                    showSignOutConfirmationDialog()
                     true
                 }
+
                 else -> {
                     navController.navigate(item.itemId)
                     true
                 }
             }
         }
     }
+
+    private fun showSignOutConfirmationDialog() {
+        AlertDialog.Builder(this)
+            .setTitle(getString(R.string.logout_title))
+            .setMessage(getString(R.string.logout_message))
+            .setPositiveButton(getString(R.string.yes)) { dialog, _ ->
+                signOut()
+                dialog.dismiss()
+            }
+            .setNegativeButton(getString(R.string.no)) { dialog, _ ->
+                dialog.dismiss()
+            }
+            .show()
+    }
+
     private fun signOut() {
-        authRepository.signOut()
-        Toast.makeText(this, "Logout successful", Toast.LENGTH_LONG).show()
+        mainViewModel.signOut()
+        Toast.makeText(this, getString(R.string.logout_successful), Toast.LENGTH_LONG).show()
         navController.navigate(R.id.signInFragment)
```

Comment:
burada aslında direk sign in fragmente gidiyorsun signOut success gelirse gitmesi gerek aslında şuan buna bakmıyor çıkış yaparken bi hata da olabilir. Burada back stack durumuna da bir bakman gerekebilir signIn fragemnte gidiyor ama arkada home açık kalmış olabilir belki bakmak gerek.